File: Python/implementation_03.py
# 2021.01.04 Implementation Exercise
# 예제 4-3 왕실의 나이트 - p.115

# 시작 위치 입력
input = input()

# 시작 행
row = int(input[1])
# 시작 열
col = int(ord(input[0]) - ord('a') + 1)

cnt = 0
if row + 2 <= 8 and col + 1 <= 8:
    cnt += 1
if row + 2 <= 8 and col - 1 >= 1:
    cnt += 1
if row - 2 >= 1 and col + 1 <= 8:
    cnt += 1
if row -2 >= 1 and col + 1 <= 8:
    cnt += 1
if row + 1 <= 8 and col + 2 <= 8:
    cnt += 1
if row + 1 <= 8 and col -2 >= 1:
    cnt += 1
if row - 1 >= 1 and col + 2 <= 8:
    cnt += 1
if row - 1 >= 1 and col - 2 >= 1:
    cnt += 1
print(cnt)        

File: Java/CodingTest/src/02_Implementation/implementation_03.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

/* 왕실의 나이트 */
public class implementation_03 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String inputString = br.readLine();
        int row = inputString.charAt(1) - '0';
        int col = inputString.charAt(0) - 'a' + 1;
        int[][] types = { { -2, -1 }, { -2, 1 }, { 2, -1 }, { 2, 1 }, { -1, -2 }, { -1, 2 }, { 1, -2 }, { 1, 2 } };
        int cnt = 0;

        int nrow, ncol;
        for (int[] type: types) {
            nrow = row + type[0];
            ncol = col + type[1];
            if (nrow < 1 || nrow > 8 || ncol < 1 || ncol > 8) {
                continue;
            }
            cnt++;
        }
        System.out.println(cnt);
    }
}
